src/abi_1_0.c

```diff
@@ -115,42 +113,14 @@ struct fi_info_1_1 {
 	void				*src_addr;
 	void				*dest_addr;
 	fid_t				handle;
-	struct fi_tx_attr		*tx_attr;
-	struct fi_rx_attr		*rx_attr;
```

Comment:
Was this a bug before?

Reply:
For the fi_tx_attr, yes.  The fi_rx_attr has not changed since the initial release.  The fi_tx_attr structure added a tclass field starting at ABI 1.3.  For ABI 1.0->1.2, the structure should have stopped with rma_iov_limit as the last field.

fi_tx_attr_1_2 should have been name fi_tx_attr_1_0 and used with the 1.0 and 1.1 ABIs.  That is, the structure changed between ABI 1.2 and 1.3, but ABI 1.2 used the same structure as 1.1 and 1.0.

Worst case is that libfabric could have read beyond the end of the data structure (see ofi_dupinfo) passed in by the user when handling compatibility.  In practice, this was unlikely to occur.  libfabric always returns the current ABI data structures from API calls, such as fi_getinfo and fi_dupinfo.  So, even though the app wasn't aware of the extra fields, the data was present